django-entrypoint.sh

```diff
@@ -1,63 +1,69 @@
 #!/usr/bin/env sh
 set -e
 
-if [ -z "$APP_MODULE" ]; then
-  echo "The \$APP_MODULE environment variable must be set to the WSGI application module name"
-  exit 1
+# No args or looks like options or the APP_MODULE for Gunicorn
+if [ "$#" = 0 ] || \
+    [ "${1#-}" != "$1" ] || \
+    echo "$1" | grep -Eq '^([_A-Za-z]\w*\.)*[_A-Za-z]\w*:[_A-Za-z]\w*$'; then
+  set -- gunicorn "$@"
 fi
 
-# Do an extra chown of the /app directory at runtime in addition to the one in
-# the build process in case any directories are mounted as root-owned volumes at
-# runtime.
-chown -R gunicorn:gunicorn /app
+# Looks like a Celery command, let's run that with Celery's entrypoint script
+if [ "$1" = 'celery' ]; then
+  set -- celery-entrypoint.sh "$@"
+fi
+
+if [ "$1" = 'gunicorn' ]; then
+  # Do an extra chown of the /app directory at runtime in addition to the one in
+  # the build process in case any directories are mounted as root-owned volumes
+  # at runtime.
+  chown -R gunicorn:gunicorn /app
 
-# Run the migration as the gunicorn user so that if it creates a local DB (e.g.
-# when using sqlite in development), that DB is still writable. Ultimately, the
-# user shouldn't really be using a local DB and it's difficult to offer support
-# for all the cases in which a local DB might be created -- but here we do the
-# minimum.
-su-exec gunicorn django-admin migrate --noinput
+  # Run the migration as the gunicorn user so that if it creates a local DB
+  # (e.g. when using sqlite in development), that DB is still writable.
+  # Ultimately, the user shouldn't really be using a local DB and it's difficult
+  # to offer support for all the cases in which a local DB might be created --
+  # but here we do the minimum.
+  su-exec gunicorn django-admin migrate --noinput
 
-if [ -n "$SUPERUSER_PASSWORD" ]; then
-  echo "from django.contrib.auth.models import User
+  if [ -n "$SUPERUSER_PASSWORD" ]; then
+    echo "from django.contrib.auth.models import User
 if not User.objects.filter(username='admin').exists():
     User.objects.create_superuser('admin', 'admin@example.com', '$SUPERUSER_PASSWORD')
 " | su-exec gunicorn django-admin shell
```

Comment:
My thought was the same.. and I tried but I couldn't get it to work. The `django-admin shell` command doesn't take `-` like `python` does.